airflow/www/static/js/dag/index.tsx

```diff
@@ -30,13 +31,24 @@ import Main from "./Main";
 // create shadowRoot
 const root = document.querySelector("#root");
 const shadowRoot = root?.attachShadow({ mode: "open" });
+
 const cache = createCache({
   container: shadowRoot,
   key: "c",
 });
 const mainElement = document.getElementById("react-container");
 
 if (mainElement) {
+  const styleTag = document.createElement("style");
+  // We need to set the z-index manually to make sure edges inside of groups still work
+  let style = reactFlowStyle.toString();
+  style += `
```

Comment:
To avoid this we could set a `background-color: transparent` on nodes that are taskgroup when expended/isOpen.

I tried playing around with z-index trying to avoid the `!important` but it looks like the stacking contexts are a bit tricky there.

Reply:
Fixed with your `defaultEdgeOptions={{ zIndex: 1 }}` suggestion.